File: magma_scripts/original_sirsforced/weights_with_trb_pos_char.m
SetNthreads(64);
Q := GF(11863279);
SetVerbose("Faugere", 2);
P<i_7, x1_6, s_6, i_6, x2_5, x1_5, s_5, r_5, i_5, x2_4, x1_4, s_4, r_4, i_4, x2_3, x1_3, s_3, r_3, i_3, x2_2, x1_2, s_2, r_2, i_2, x2_1, x1_1, s_1, r_1, i_1, x2_0, x1_0, s_0, r_0, i_0, z_aux, w_aux, M, b0, b1, g, mu, nu>:= PolynomialRing(Q, 42, "grevlex");
G := ideal< P | 10716553494454577480-i_0, -b0*b1*i_0*s_0^2*x1_0^2-b0*i_0*s_0^2+i_0*mu+i_0*nu+i_1, 27942744439186952753-r_0, g*r_0-i_0*nu+mu*r_0+r_1, -i_1+17857409235577475769406404975290073300278762639302358642394319500748836454316568947396972498800, ((-b1*x1_0^2-1)*s_0^2*i_1-(b1*s_0^2*x1_1^2+b1*s_1^2*x1_0^2+s_1^2)*i_0)*b0+(nu+mu)*i_1+i_2, b0*b1*i_0*s_0^2*x1_0^2+b0*i_0*s_0^2+mu*s_0^2-g*r_0+s_1^2-mu, M^3*x2_0^3+x1_1^2, -r_1-539290885820445506472503010670159515495, -nu*i_1+(mu+g)*r_1+r_2, -i_2-14151031524407262684236429214307210470195892059174767003787625738956306362740260242599682080118987666819019075115033927742885654328486982055651633959834308922812694164520, (((-s_0^2*x1_2^2-2*s_1^2*x1_1^2-s_2^2*x1_0^2)*i_0-x1_0^2*i_2*s_0^2-2*i_1*(s_0^2*x1_1^2+s_1^2*x1_0^2))*b1-2*s_1^2*i_1-s_2^2*i_0-i_2*s_0^2)*b0+(nu+mu)*i_2+i_3, ((b1*s_0^2*x1_1^2+b1*s_1^2*x1_0^2+s_1^2)*i_0+i_1*s_0^2*(b1*x1_0^2+1))*b0-g*r_1+mu*s_1^2+s_2^2, M^3*x2_1^3+x1_2^2, -M^3*x1_0^2+x2_1^3, -r_2+497946715908217671756532106978922968522881179069692836036732344859843655233883874919622205735687211971293674948425, -nu*i_2+(mu+g)*r_2+r_3, -i_3-135115703586344426784164658808391195989095939971984904372866576463786267720849775146849544438870968161781575955610798357707913340830633344781391088200997409456097421370137596899269345116179950021462975917657604526320966003866076618855010514748440, (((-s_0^2*x1_3^2-3*s_1^2*x1_2^2-3*s_2^2*x1_1^2-s_3^2*x1_0^2)*i_0+(-3*i_1*x1_2^2-3*i_2*x1_1^2-i_3*x1_0^2)*s_0^2+(-3*i_1*s_2^2-3*i_2*s_1^2)*x1_0^2-6*s_1^2*x1_1^2*i_1)*b1-s_0^2*i_3-3*s_1^2*i_2-3*s_2^2*i_1-s_3^2*i_0)*b0+(nu+mu)*i_3+i_4, (((s_0^2*x1_2^2+2*s_1^2*x1_1^2+s_2^2*x1_0^2)*i_0+(2*i_1*x1_1^2+i_2*x1_0^2)*s_0^2+2*s_1^2*x1_0^2*i_1)*b1+2*s_1^2*i_1+s_2^2*i_0+i_2*s_0^2)*b0-g*r_2+mu*s_2^2+s_3^2, M^3*x2_2^3+x1_3^2, -M^3*x1_1^2+x2_2^3, -i_4+454946859466174366088565328125741558730662636632841004998439019918614870187663167017733307015834086370541282682472138556867920070711769098917322812039575950587311812779916348688199269042266785149309993952667253529541646773523287066866551594462387205343775006880920089809369552294954972761792335443274435800303694255998480, (((-s_0^2*x1_4^2-4*s_1^2*x1_3^2-6*s_2^2*x1_2^2-4*s_3^2*x1_1^2-s_4^2*x1_0^2)*i_0+(-4*i_1*x1_3^2-6*i_2*x1_2^2-4*i_3*x1_1^2-i_4*x1_0^2)*s_0^2+(-4*i_1*s_3^2-6*i_2*s_2^2-4*i_3*s_1^2)*x1_0^2+(-12*s_1^2*x1_2^2-12*s_2^2*x1_1^2)*i_1-12*s_1^2*x1_1^2*i_2)*b1-s_0^2*i_4-4*s_1^2*i_3-6*s_2^2*i_2-4*s_3^2*i_1-s_4^2*i_0)*b0+(nu+mu)*i_4+i_5, (((s_0^2*x1_3^2+3*s_1^2*x1_2^2+3*s_2^2*x1_1^2+s_3^2*x1_0^2)*i_0+(3*i_1*x1_2^2+3*i_2*x1_1^2+i_3*x1_0^2)*s_0^2+(3*i_1*s_2^2+3*i_2*s_1^2)*x1_0^2+6*s_1^2*x1_1^2*i_1)*b1+s_0^2*i_3+3*s_1^2*i_2+3*s_2^2*i_1+s_3^2*i_0)*b0-g*r_3+mu*s_3^2+s_4^2, M^3*x2_3^3+x1_4^2, -M^3*x1_2^2+x2_3^3, -i_5+3792541593000254720245825801865142876060081923847829450878974729919518778951398456326191507529084614846211415834441272759707383968402385605288983375997407728074453956149517241338027672045739099143442167440138012617647753694127083821047227305372255486027261539962036034222649575031818871080706984355949015545486606694112771095580561921811755022989843401474967018524129464358360748390075746415761320, (((-s_0^2*x1_5^2-5*s_1^2*x1_4^2-10*s_2^2*x1_3^2-10*s_3^2*x1_2^2-5*s_4^2*x1_1^2-s_5^2*x1_0^2)*i_0+(-5*i_1*x1_4^2-10*i_2*x1_3^2-10*i_3*x1_2^2-5*i_4*x1_1^2-i_5*x1_0^2)*s_0^2+(-5*i_1*s_4^2-10*i_2*s_3^2-10*i_3*s_2^2-5*i_4*s_1^2)*x1_0^2+(-20*s_1^2*x1_3^2-30*s_2^2*x1_2^2-20*s_3^2*x1_1^2)*i_1+(-30*i_2*x1_2^2-20*i_3*x1_1^2)*s_1^2-30*s_2^2*x1_1^2*i_2)*b1-i_5*s_0^2-5*s_1^2*i_4-10*s_2^2*i_3-10*s_3^2*i_2-5*s_4^2*i_1-s_5^2*i_0)*b0+(nu+mu)*i_5+i_6, (((s_0^2*x1_4^2+4*s_1^2*x1_3^2+6*s_2^2*x1_2^2+4*s_3^2*x1_1^2+s_4^2*x1_0^2)*i_0+(4*i_1*x1_3^2+6*i_2*x1_2^2+4*i_3*x1_1^2+i_4*x1_0^2)*s_0^2+(4*i_1*s_3^2+6*i_2*s_2^2+4*i_3*s_1^2)*x1_0^2+(12*s_1^2*x1_2^2+12*s_2^2*x1_1^2)*i_1+12*s_1^2*x1_1^2*i_2)*b1+s_0^2*i_4+4*s_1^2*i_3+6*s_2^2*i_2+4*s_3^2*i_1+s_4^2*i_0)*b0-g*r_4+mu*s_4^2+s_5^2, M^3*x2_4^3+x1_5^2, -nu*i_3+(mu+g)*r_3+r_4, -M^3*x1_3^2+x2_4^3, -i_6-30419155584988490377335329888142829119869746118054306192428448552704103498021294240168621902269120524767534893189419971985125632549634673862034102283504948906739611232193896944495666311396059275165226006007225682363933876736069438049683736731292165567970304606511258155896898799699887977117705914067104051022185488342121634329412276134042852645747498330084926804822206488276503958122576348370888369913937306125337616491325738437092791335196579691156593450103504040457079320, (((-s_0^2*x1_6^2-6*s_1^2*x1_5^2-15*s_2^2*x1_4^2-20*s_3^2*x1_3^2-15*s_4^2*x1_2^2-6*s_5^2*x1_1^2-s_6^2*x1_0^2)*i_0+(-6*i_1*x1_5^2-15*i_2*x1_4^2-20*i_3*x1_3^2-15*i_4*x1_2^2-6*i_5*x1_1^2-i_6*x1_0^2)*s_0^2+(-6*i_1*s_5^2-15*i_2*s_4^2-20*i_3*s_3^2-15*i_4*s_2^2-6*i_5*s_1^2)*x1_0^2+(-30*s_1^2*x1_4^2-60*s_2^2*x1_3^2-60*s_3^2*x1_2^2-30*s_4^2*x1_1^2)*i_1+(-60*i_2*x1_3^2-60*i_3*x1_2^2-30*i_4*x1_1^2)*s_1^2+(-60*i_2*s_3^2-60*i_3*s_2^2)*x1_1^2-90*s_2^2*x1_2^2*i_2)*b1-6*i_5*s_1^2-i_6*s_0^2-15*s_2^2*i_4-20*s_3^2*i_3-15*s_4^2*i_2-6*s_5^2*i_1-s_6^2*i_0)*b0+(nu+mu)*i_6+i_7, (((s_0^2*x1_5^2+5*s_1^2*x1_4^2+10*s_2^2*x1_3^2+10*s_3^2*x1_2^2+5*s_4^2*x1_1^2+s_5^2*x1_0^2)*i_0+(5*i_1*x1_4^2+10*i_2*x1_3^2+10*i_3*x1_2^2+5*i_4*x1_1^2+i_5*x1_0^2)*s_0^2+(5*i_1*s_4^2+10*i_2*s_3^2+10*i_3*s_2^2+5*i_4*s_1^2)*x1_0^2+(20*s_1^2*x1_3^2+30*s_2^2*x1_2^2+20*s_3^2*x1_1^2)*i_1+(30*i_2*x1_2^2+20*i_3*x1_1^2)*s_1^2+30*s_2^2*x1_1^2*i_2)*b1+i_5*s_0^2+5*s_1^2*i_4+10*s_2^2*i_3+10*s_3^2*i_2+5*s_4^2*i_1+s_5^2*i_0)*b0-g*r_5+mu*s_5^2+s_6^2, M^3*x2_5^3+x1_6^2, -nu*i_4+(mu+g)*r_4+r_5, -M^3*x1_4^2+x2_5^3, -i_7-201818786732742519958202239387631002709047842825965808863606000211587338302017344910695499615631046739342323288591591560489567180815062131476232213425087078167259695497354147990653869506812846287697079130743690760676331005668988774736691240803536020197966411816223223801142690656075927638504616644516216283226943259406789004693658666763385843434082258150681318338334364489964921649350275214820758598105973129884130316129877555911534400224141649108559227400575248797105930047688333741882576096729122829852579529159152794404336834933381173242962617120, -r_3-394595855498093821589552956766400004960743241549772042930533360784153562575680785177036500045997828093045913219665626666698362588374803513661550152872542588593216125341718959723544075770495, -r_4-3767647365912688669095498363870365701330659522923571033455895640631836147336117205519992409924953897447398700586210746736865023574255148085016450158120748166870538948576128935335303769611526151283580592674366683332037434479994458789297564323719372694681257175073015, -r_5+12686011257030650256975617040127544686943167207460441645009959441668064591653201359974439860791368401290541911668191290052874661389526840115743855076604083392047703615516126915696225883586714126376306474034743922013877200449968472579409474746005342261150653999748109384596600530855689457994737582220805954536094201323658814986745338916492505, z_aux^2-1>;
time GroebnerBasis(G);// {M = M^3, i_0 = i_0, i_1 = i_1, i_2 = i_2, i_3 = i_3, i_4 = i_4, i_5 = i_5, i_6 = i_6, i_7 = i_7, r_0 = r_0, r_1 = r_1, r_2 = r_2, r_3 = r_3, r_4 = r_4, r_5 = r_5, s_0 = s_0^2, s_1 = s_1^2, s_2 = s_2^2, s_3 = s_3^2, s_4 = s_4^2, s_5 = s_5^2, s_6 = s_6^2, x1_0 = x1_0^2, x1_1 = x1_1^2, x1_2 = x1_2^2, x1_3 = x1_3^2, x1_4 = x1_4^2, x1_5 = x1_5^2, x1_6 = x1_6^2, x2_0 = x2_0^3, x2_1 = x2_1^3, x2_2 = x2_2^3, x2_3 = x2_3^3, x2_4 = x2_4^3, x2_5 = x2_5^3, z_aux = z_aux^2}
quit;

File: julia_scripts/new-goodwin-osc/weights_with_trb_pos_char.jl
using Oscar
R, vars = PolynomialRing(FiniteField(11863279), "x1_9, x4_8, x1_8, x4_7, x3_7, x2_7, x1_7, x4_6, x3_6, x2_6, x1_6, x4_5, x3_5, x2_5, x1_5, x4_4, x3_4, x2_4, x1_4, x4_3, x3_3, x2_3, x1_3, x4_2, x3_2, x2_2, x1_2, x4_1, x3_1, x2_1, x1_1, x4_0, x3_0, x2_0, x1_0, z_aux, w_aux, alpha, b, beta, c, delta, gama, sgm")
I = ideal(R, [8385335915922217088-x1_0, b*x1_0*x4_0^2+b*c*x1_0+x1_1*x4_0^2+c*x1_1-1, -x1_1-133429835018152541671530516253661646457390138564648913151/8587410195430989983, ((x4_0^2+c)*b+x4_1^2)*x1_1+b*x1_0*x4_1^2+(x4_0^2+c)*x1_2, delta*sgm*x3_0^3*x4_0^2-gama*sgm*x2_0^3*x4_0^2+x3_0^3*x4_1^2, -x1_2+9264645191930246386980550746901609770361545569971552418834689790657868975820724939141205345535436544874620938378/321786467028463463965878352756161089108160561803208208579, ((x4_0^2+c)*x1_2+x1_0*x4_2^2+2*x1_1*x4_1^2)*b+2*x1_2*x4_1^2+x1_1*x4_2^2+(x4_0^2+c)*x1_3, ((delta*x3_0^3-gama*x2_0^3)*x4_1^2+x4_0^2*(delta*x3_1^3-gama*x2_1^3))*sgm+x3_0^3*x4_2^2+x3_1^3*x4_1^2, beta^4*x2_0^3+x2_1^3-alpha*x1_0, delta*x3_0^3-gama*x2_0^3+x3_1^3, -x1_3-643286792048262228282392185685651999513551130178613461290669741180423778477561696702429775908364731070198628279946190486442426776895110487097120150472989618595299341020/12057946226646222293087585139886798525326103031359751559117039744485449578719886441103835795927, ((x4_0^2+c)*x1_3+3*x1_1*x4_2^2+x4_3^2*x1_0+3*x1_2*x4_1^2)*b+3*x1_3*x4_1^2+x1_1*x4_3^2+3*x1_2*x4_2^2+(x4_0^2+c)*x1_4, ((x3_0^3*x4_2^2+2*x3_1^3*x4_1^2+x3_2^3*x4_0^2)*delta-2*(x4_1^2*x2_1^3+1/2*x4_2^2*x2_0^3+1/2*x2_2^3*x4_0^2)*gama)*sgm+x3_2^3*x4_1^2+2*x3_1^3*x4_2^2+x3_0^3*x4_3^2, beta^4*x2_1^3+x2_2^3-alpha*x1_1, delta*x3_1^3-gama*x2_1^3+x3_2^3, -x1_4+4283306374775936788706524451043965387430293412477638104828929598097474408313139701689236224957056093579319731486421648958719946381693999772036781556459235518924839805231832455580199240821667186411388564743803091660550485275176/451834002055254578374000685462339908213287864105566716485569675289681197707900438381069277065822300475737761084229799022463767284251, ((x4_0^2+c)*x1_4+4*x1_3*x4_1^2+6*x1_2*x4_2^2+4*x1_1*x4_3^2+x4_4^2*x1_0)*b+4*x1_4*x4_1^2+x1_1*x4_4^2+4*x1_2*x4_3^2+6*x1_3*x4_2^2+(x4_0^2+c)*x1_5, ((x3_0^3*x4_3^2+3*x3_1^3*x4_2^2+3*x3_2^3*x4_1^2+x3_3^3*x4_0^2)*delta-gama*(x2_0^3*x4_3^2+3*x2_1^3*x4_2^2+3*x2_2^3*x4_1^2+x2_3^3*x4_0^2))*sgm+x3_3^3*x4_1^2+3*x3_2^3*x4_2^2+3*x3_1^3*x4_3^2+x3_0^3*x4_4^2, beta^4*x2_2^3+x2_3^3-alpha*x1_2, delta*x3_2^3-gama*x2_2^3+x3_3^3, -x1_5-1666677288025800409145529540076836115904551098339125826184757551594244676372447463284591612925079680274228461875672169791057477039564246800318147156797756959720601854882280040602409636012676638624919103353991973068195901048863900626391579111139018780585674058155462066035343684889903657744908749540464/16931072802607020173587418519919868218582373929712273411482981907350312395143462053346286067526200145819400128000256042614602595579915777531317482946301100985968433798863, ((x4_0^2+c)*x1_5+5*x1_4*x4_1^2+10*x1_3*x4_2^2+10*x1_2*x4_3^2+5*x1_1*x4_4^2+x4_5^2*x1_0)*b+5*x1_5*x4_1^2+x1_1*x4_5^2+5*x1_2*x4_4^2+10*x1_3*x4_3^2+10*x1_4*x4_2^2+(x4_0^2+c)*x1_6, ((x3_0^3*x4_4^2+4*x3_1^3*x4_3^2+6*x3_2^3*x4_2^2+4*x3_3^3*x4_1^2+x3_4^3*x4_0^2)*delta-gama*(x2_0^3*x4_4^2+4*x2_1^3*x4_3^2+6*x2_2^3*x4_2^2+4*x2_3^3*x4_1^2+x2_4^3*x4_0^2))*sgm+6*x3_2^3*x4_3^2+4*x3_3^3*x4_2^2+x3_4^3*x4_1^2+4*x3_1^3*x4_4^2+x3_0^3*x4_5^2, beta^4*x2_3^3+x2_4^3-alpha*x1_3, delta*x3_3^3-gama*x2_3^3+x3_4^3, -x1_6-33608713638191382491188793862329717670544397057496053433179189646888498775050052059334820942325456724354557429984817246685539408765937814529826463672107351622457122559717982039043206312306204679128317263275891179871720955380709453250146173464586149238753103294607870690483228604769358547309536053431960160838222034438815766757613655082446913882754016720542974200096812694615904/634439251900576228300915347863807891139692858904157559681402011424374272116396614206945174702457143409194876567401702958024295571316184498535659939950197870357442650908527535104819755030017506910544260692019, ((x4_0^2+c)*x1_6+6*x1_5*x4_1^2+15*x1_4*x4_2^2+20*x1_3*x4_3^2+15*x1_2*x4_4^2+6*x1_1*x4_5^2+x4_6^2*x1_0)*b+6*x1_6*x4_1^2+x1_1*x4_6^2+6*x1_2*x4_5^2+15*x1_3*x4_4^2+20*x1_4*x4_3^2+15*x1_5*x4_2^2+(x4_0^2+c)*x1_7, ((x3_0^3*x4_5^2+5*x3_1^3*x4_4^2+10*x3_2^3*x4_3^2+10*x3_3^3*x4_2^2+5*x3_4^3*x4_1^2+x3_5^3*x4_0^2)*delta-gama*(x2_0^3*x4_5^2+5*x2_1^3*x4_4^2+10*x2_2^3*x4_3^2+10*x2_3^3*x4_2^2+5*x2_4^3*x4_1^2+x2_5^3*x4_0^2))*sgm+5*x3_1^3*x4_5^2+10*x3_2^3*x4_4^2+10*x3_3^3*x4_3^2+5*x3_4^3*x4_2^2+x3_5^3*x4_1^2+x3_0^3*x4_6^2, beta^4*x2_4^3+x2_5^3-alpha*x1_4, delta*x3_4^3-gama*x2_4^3+x3_5^3, -x1_7-116265740233809041818393698144185154791876944220167630827048119984464080213872222311732719147200378166194034113528597321334463753239283021979472546223559620441841572939325858435911300359011221986464528954147611830742881019882448739660473639586430434634153085183993583281017230293229925868360900469856111585003927783025714100493136048005948726580708344187528155326107452025071590857812961821795164626801908803696606758975509265523829247671755864094713792/23773636144910112741645368476615933391617934604996028585282565650101293343891417519645503958369724860620051923141919631167176208902376837177298275001709702158531941328026877377085751765009275107606340280313331573397177590080070469032045610464647, ((x4_0^2+c)*x1_7+7*x1_6*x4_1^2+21*x1_5*x4_2^2+35*x1_4*x4_3^2+35*x1_3*x4_4^2+21*x1_2*x4_5^2+7*x1_1*x4_6^2+x4_7^2*x1_0)*b+7*x1_7*x4_1^2+x1_1*x4_7^2+7*x1_2*x4_6^2+21*x1_3*x4_5^2+35*x1_4*x4_4^2+35*x1_5*x4_3^2+21*x1_6*x4_2^2+(x4_0^2+c)*x1_8, ((x3_0^3*x4_6^2+6*x3_1^3*x4_5^2+15*x3_2^3*x4_4^2+20*x3_3^3*x4_3^2+15*x3_4^3*x4_2^2+6*x3_5^3*x4_1^2+x3_6^3*x4_0^2)*delta-gama*(x2_0^3*x4_6^2+6*x2_1^3*x4_5^2+15*x2_2^3*x4_4^2+20*x2_3^3*x4_3^2+15*x2_4^3*x4_2^2+6*x2_5^3*x4_1^2+x2_6^3*x4_0^2))*sgm+x3_0^3*x4_7^2+6*x3_1^3*x4_6^2+15*x3_2^3*x4_5^2+20*x3_3^3*x4_4^2+15*x3_4^3*x4_3^2+6*x3_5^3*x4_2^2+x3_6^3*x4_1^2, beta^4*x2_5^3+x2_6^3-alpha*x1_5, delta*x3_5^3-gama*x2_5^3+x3_6^3, -x1_8-9585957682104606876693730189533622594428107906156973705014641774225536999049165475342685155206199863154431195804914709407123912611589683681844243351507488023409646687759680101329825554885478541047640348449711157021781407782653844431205577593652448251467892061048217947962498944339676335633282678488168184083884417572738937728887168737988638329725822710217097021773685720205066649892110560819873995012043416414842238911363354924603471135405080821114477514153402591443154735361518377736735057929328534682295590785102320316149376/127263287756519187378776672211963283328744941869146797710843069272187023607768841705782620053478768027698348141438432846787595967077167983008607228713740782244336768347690945763071728990069043431321463943154482938667971298546577483042295574347865272770482000553037062467055974481373, ((x4_0^2+c)*x1_8+8*x1_7*x4_1^2+28*x1_6*x4_2^2+56*x1_5*x4_3^2+70*x1_4*x4_4^2+56*x1_3*x4_5^2+28*x1_2*x4_6^2+8*x1_1*x4_7^2+x4_8^2*x1_0)*b+8*x1_8*x4_1^2+x1_1*x4_8^2+8*x1_2*x4_7^2+28*x1_3*x4_6^2+56*x1_4*x4_5^2+70*x1_5*x4_4^2+56*x1_6*x4_3^2+28*x1_7*x4_2^2+(x4_0^2+c)*x1_9, ((x3_0^3*x4_7^2+7*x3_1^3*x4_6^2+21*x3_2^3*x4_5^2+35*x3_3^3*x4_4^2+35*x3_4^3*x4_3^2+21*x3_5^3*x4_2^2+7*x3_6^3*x4_1^2+x3_7^3*x4_0^2)*delta-gama*(x2_0^3*x4_7^2+7*x2_1^3*x4_6^2+21*x2_2^3*x4_5^2+35*x2_3^3*x4_4^2+35*x2_4^3*x4_3^2+21*x2_5^3*x4_2^2+7*x2_6^3*x4_1^2+x2_7^3*x4_0^2))*sgm+x3_0^3*x4_8^2+7*x3_1^3*x4_7^2+21*x3_2^3*x4_6^2+35*x3_3^3*x4_5^2+35*x3_4^3*x4_4^2+21*x3_5^3*x4_3^2+7*x3_6^3*x4_2^2+x3_7^3*x4_1^2, beta^4*x2_6^3+x2_7^3-alpha*x1_6, delta*x3_6^3-gama*x2_6^3+x3_7^3, -x1_9+256224927275578405615783144394079807716950410863946612053346418896072875249947827195920624131192404159399161825138147699722310534012823278896922655587304497918949606160044886115802822134703797262472207006546801215647814836909611438238991627933027167507451208213557951198572978820439528928741394247551802170855706915127184058674249143567894243011232073378653119012526894786140743317185316415874715191603413067333613582153094247958838536862098075671133226131008945207273838403621597955579013412311753195184139996870236095471231672306197271499859570646633926516486310872763732715922278453566923113641312000/4768795576038244252674508426535068878796731286939242265445732098141700408290738306904037492226385745192250003727335015884872691760054383703323056845036914328896256496306485744697387608273649950578920242340228734058164939735536386865779292044225233394415822572462890952742738902488517247724167383822366640824273104600649, z_aux^3*(x4_0^2+c)*x3_0^3-1])
gb = f4(I)
# {beta = beta^4, x1_0 = x1_0, x1_1 = x1_1, x1_2 = x1_2, x1_3 = x1_3, x1_4 = x1_4, x1_5 = x1_5, x1_6 = x1_6, x1_7 = x1_7, x1_8 = x1_8, x1_9 = x1_9, x2_0 = x2_0^3, x2_1 = x2_1^3, x2_2 = x2_2^3, x2_3 = x2_3^3, x2_4 = x2_4^3, x2_5 = x2_5^3, x2_6 = x2_6^3, x2_7 = x2_7^3, x3_0 = x3_0^3, x3_1 = x3_1^3, x3_2 = x3_2^3, x3_3 = x3_3^3, x3_4 = x3_4^3, x3_5 = x3_5^3, x3_6 = x3_6^3, x3_7 = x3_7^3, x4_0 = x4_0^2, x4_1 = x4_1^2, x4_2 = x4_2^2, x4_3 = x4_3^2, x4_4 = x4_4^2, x4_5 = x4_5^2, x4_6 = x4_6^2, x4_7 = x4_7^2, x4_8 = x4_8^2, z_aux = z_aux^3}